Dimmer.h

```diff
@@ -1,73 +1,132 @@
 
-class Dimmer{
-    
-public:
-    Dimmer();
-    void attachZeroCross(int zc_pin, int inter);
-    void attachTriac(byte pin);
-    int init();
-    int initRamp();
-    int initCount(byte resolution = 1);
+/**
+ * Possible operation modes for the dimmer library.
+ */
+#define NORMAL_MODE     0
+#define RAMP_MODE       1
+#define COUNT_MODE      2
+
+/**
+ * Possible light state for the dimmer library.
+ */
+#define OFF    0
+#define ON     1
+
+/**
+ * A ZeroCross Nanoshield or similar module to dimm AC lights.
+ */
+class DimmerControl
+{    
+  public:
+  	/**
+     * Constructor.
+     *
+     * Creates an object to access one Triac Nanoshield.
+     * @param triacPin	pin select matching the jumper on the board. (D3, D5, etc)
+     * @param value 	initial intensity in percentage of the dimm light.
+     *					Minimum is 0 and maximum is 100. Defaut is 50%.
+     * @param state 	initial state of the light. Possible states: ON or OFF. Default is OFF.
+     * @see   begin()
+     */ 
+    DimmerControl(uint8_t triacPin, uint8_t value=50, bool state = OFF);
+
+    /**
+     * Initializes the module.
+     *
+     * Initializes ZeroCross and Timer 2 interrupts. Set the light according to initial settings.
+     * @param mode 		operation mode to crontrol the light. 
+     *					Possible modes:
+     *					NORMAL_MODE: Uses timer to apply only a percentage of the AC power to the light.
+     *					RAMP_MODE: Same as in normal mode, but it applies a ramp effect to the light.
+     *					COUNT_MODE: Counts AC waves and applies full waves from time to time. @see resolution
+	 * @param resolution 	IN COUNT MODE:
+     *                          divided prescalar to define how many pulses to count.
+	 *                          if resolution is 1 the mode counts 100 pulses, if resolution is 2
+	 *                          the mode counts 50 pulses, and so on.
+     *
+     *                      IN RAMP MODE:
+	 *                          Controlls the speed of the ramp when changing values.
+     */
+    bool begin(uint8_t mode = NORMAL_MODE, uint8_t resolution = 1);
+
+    /**
+     * Turns the light OFF. 
+     */
     void off();
-    void off(byte lamp);
+
+    /**
+     * Turns the light ON. 
+     */
     void on();
-    void on(byte lamp);
-    void toggle(byte lamp);
-    void set(byte lamp, byte value);
+
+    /**
+     * Toggles the light state. 
+     */
+    void toggle();
+
+    /**
+     * Gets the current value of the dimmed light.
+     *
+     * @return current light value.
+     */
+    bool getState();
+
+    /**
+     * Gets the current state of the dimmed light.
+     *
+     * @return current light state.
+     */
+    uint8_t getValue();
+
+    /**
+     * Sets the value and the state of the light.
+     * @param value 	the value of the dimm light. Values from 0 to 100.
+     * @param state 	the state of the light. ON or OFF.
+     */
+    void set(uint8_t value, uint8_t state = -1);
+
     void zeroCross();
```

Comment:
Ideally, we would want to make `zeroCross()` and `triac()` private as they should never be called by a user of the class. However, this would cause a few problems since they must be called from the ISRs.

One way to solve this is to create two functions inside the `Dimmer.cpp`: `callTriac(int i)` and `callZeroCross()` (the latter already exists). The Timer 2 ISR and `attachInterrupt()` would call these functions, and they would subsequently call the private methods of the `Dimmer` class.

To be able to call these private methods, `callTriac(int i)` and `callZeroCross()` must be made friend functions of the `Dimmer` class. An example of how to do this is shown in this tutorial: http://www.tutorialspoint.com/cplusplus/cpp_friend_functions.htm

We should do with `callTriac(int i)` and `callZeroCross()` the same thing that is done with `printWidth()` in the tutorial.